java-components/build-request-processor/src/test/java/com/redhat/hacbs/container/analyser/deploy/git/GitTest.java

```diff
@@ -68,11 +70,13 @@ public void testPush()
             FileUtils.copyDirectory(new File(repoRoot.toFile(), ".git"), new File(initialRepo.toFile(), ".git"));
             if (initialRepository.tagList().call().stream().noneMatch(r -> r.getName().equals("refs/tags/0.1"))) {
                 // Don't have the tag and cannot guarantee a fork will have it so fetch from primary repository.
+                String newRemote = RandomStringUtils.randomAlphabetic(8);
+                Log.infof("Current repo does not have 0.1 tag; configuring %s to fetch it.", newRemote);
                 initialRepository.remoteAdd()
-                        .setUri(new URIish("https://github.com/redhat-appstudio/jvm-build-service.git")).setName("upstream")
+                        .setUri(new URIish("https://github.com/redhat-appstudio/jvm-build-service.git")).setName(newRemote)
                         .call();
                 initialRepository.fetch().setRefSpecs("refs/tags/0.1:refs/tags/0.1").setTagOpt(TagOpt.NO_TAGS)
-                        .setRemote("upstream").call();
+                        .setRemote(newRemote).call();
```

Comment:
Annoyingly Eclipse silently hides the problem if there is an existing remote with same name - hence using a random string. We only need it to retrieve the tag anyway.

Reply:
How about adding something like this just in case even though we're using random:

```java
var existingRemotes = config.getSubsections("remote");
var newRemote = randomAlphanumeric(8);
assertThat(existingRemotes).doesNotContain(remote);
```